pyiron_base/_tests.py

```diff
@@ -27,7 +27,32 @@
 __date__ = "Mar 23, 2021"
 
 
-class TestWithProject(unittest.TestCase, ABC):
+class PyironTestCase(unittest.TestCase, ABC):
+
+    """
+    Tests that also include testing the docstrings in the specified modules
+    """
+
+    @property
+    def docstring_module(self):
+        """
+        Define module whose docstrings will be tested
+        """
+        return None
+
+    def test_docstrings(self):
+        """
+        Fails with output if docstrings in the given module fails.
+
+        Output capturing adapted from https://stackoverflow.com/a/22434594/12332968
+        """
+        with StringIO() as buf, redirect_stdout(buf):
+            result = doctest.testmod(self.docstring_module)
+            output = buf.getvalue()
+        self.failIf(result.failed > 0, msg=output)
+
+
+class TestWithProject(PyironTestCase):
```

Comment:
```suggestion
class TestWithProject(PyironTestCase, ABC):
```

We still expect this to be a parent for individual testing suites.